src/textual/widget.py

```diff
@@ -3234,19 +3234,23 @@ def watch_has_focus(self, value: bool) -> None:
         """Update from CSS if has focus state changes."""
         self._update_styles()
 
-    def watch_disabled(self) -> None:
+    def watch_disabled(self, disabled: bool) -> None:
         """Update the styles of the widget and its children when disabled is toggled."""
         from .app import ScreenStackError
 
+        if disabled and self.mouse_over:
+            self._message_queue.put_nowait(events.Leave())
```

Comment:
This block all makes sense to me other than posting the `Leave` event, because the mouse has not left the widget.

Reply:
It's because when a widget is disabled, it no longer receives Enter and Leave events. So a widget that is hovered and programmatically disabled will have received an Enter, but will never get a corresponding Leave. So whatever the widget does on enter is stuck even if the cursor is no longer over the widget.

Sending the Leave here ensures that the widget goes back to its unhovered state when disabled.

Reply:
Makes sense after the explanation in the first paragraph, thanks. Probably worth a small update to the Leave docstring.

Reply:
Done